### PR TITLE
A comparison of webhook libraries

### PR DESCRIPTION
**Pull request by wallrj**
Required by #71 

See:
 * controller-runtime webhook spike: #132 
 * openshift webhook spike: #87

### CODE REVIEW

**Comment by sebbonnet** [comment]
Required by #71

File: docs/enhancements/71-webhook.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+Openshift:
+
+The Openshift library makes it simple to also have the webhook server authenticate the requests it receives from the Kubernetes API servers.
+It takes advantage of the well established mutual authentication mechanism for aggregate API servers. See https://github.com/openshift/generic-admission-server#architecture
+
+Additionally the Openshift library performs authorization of incoming requests.
+It POSTs SubjectAccessReview API resources to the Kubernetes API server,
+asking whether the user who made the original API request is permitted to interact with the webhook server endpoint.
+For this reason, the webhook service account needs a role binding to the `system:auth-delegator` cluster role.
```
Comment (sebbonnet) [inline docs/enhancements/71-webhook.md]: According to https://kubernetes.io/docs/tasks/access-kubernetes-api/setup-extension-api-server/, an additional `extension-apiserver-authentication-reader` role is also required for accessing the `extension-apiserver-authentication` configmap.

Comment (sebbonnet) [reply docs/enhancements/71-webhook.md]: According to https://kubernetes.io/docs/tasks/access-kubernetes-api/configure-aggregation-layer/#kubernetes-apiserver-client-authentication, there are additional flags required to allow the apiserver to communicate with the extended api-server. Namely `proxy-client-*-file` and `--requestheader-*`.

Comment (wallrj) [reply docs/enhancements/71-webhook.md]: I think those flags are configure by default by most depllyment tools: https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates

**Comment by wallrj** [comment]
I've added a section on Testability and a new item on the API server flags required by the Openshift aggregate API server approach.

**Comment by sebbonnet** [comment]
I've had some more feedback from @balooo who thought that Openshift was depicted in this doc as the better solution. This made me think that we're probably not explaining clearly enough that the 2 libraries implements different types of webhooks and that this is really what the comparison should be about.  

In my opinion the choice of library is less important than the webhook model we want to follow.
To me the choice is between a simple http server webhook and an extended API server.
I feel that the former is, albeit less powerful, a well known concept and would be easier for users to reason about / setup and so to adapt. Besides it is already used in the open SDK and is likely to gain more momentum. The unstability of the library is not a huge concern as there isn't a great deal of features we use and the library is fundamentally a thin wrapper around an https server.
For these reasons my choice would go for the simple http server and so the controller-runtime library.

**Pull request merged**